### ISSUE TITLE
Mask is does'nt showing the first time

### ISSUE THREAD

**Issue by Djibdjib** [sekoyo/react-image-crop#579]
Hi and thx for this tool !

When i open the crop in a modal, the mask is not showed the first time. If i close the modal and re open, all is perfect...

Thx for your help.

**Comment by Djibdjib** [comment]
Ok i found the solution, i use the onLoad attribute on the image to set the crop values. It works !

**Comment by immanu10** [comment]
@Djibdjib I'm facing the same issue even though I'm using the **onLoad** attribute. Do you mind sharing what exactly you did to fix it?

**Comment by Djibdjib** [comment]
@immanu10 Yeah of course !

```javascript
const onImageLoad = () => {
        setCrop({
            unit: "%",
            width: currentFile.crop.w || 50,
            height: currentFile.crop.h || 50,
            x: currentFile.crop.x || 25,
            y: currentFile.crop.y || 25,
        });
    };
```
```javascript
<ReactCrop
                key={crop}
                crop={crop}
                onChange={(c, p) => onChangeCrop(c, p)}
                className="inline-block rounded-lg"
                keepSelection={true}
                ruleOfThirds={true}
            >
                <img src={currentSrcImage} className="rounded-t-lg" onLoad={onImageLoad} />
            </ReactCrop>
```

**Comment by yuhao-wong** [comment]
Adding some context to this - setting the crop onImageLoad didn't work for me. 
The crop dimensions will be correctly set, but the mask itself will be broken.

It seems like the <rect/> inside the mask SVG has 0x0 dimensions. 
I was able to force refresh it on load by doing
`document.querySelector(".ReactCrop__crop-mask")?.setAttribute("height", "100%")`

I'm curious if there is a better way around this. I am unable to consistently reproduce this issue.
![image](https://github.com/sekoyo/react-image-crop/assets/156033014/4ec7e600-0dfb-4a04-8cb3-4ab17cf04c98)

**Comment by sekoyo** [comment]
> Adding some context to this - setting the crop onImageLoad didn't work for me. The crop dimensions will be correctly set, but the mask itself will be broken.
> 
> If you inspect the DOM when this happens, it seems like the of the mask SVG has 0x0 dimensions. I was able to force refresh it on load by doing `document.querySelector(".ReactCrop__crop-mask")?.setAttribute("height", "100%")`
> 
> I'm curious if there is a better way around this. I am unable to consistently reproduce this issue. ![image](https://private-user-images.githubusercontent.com/156033014/304849780-4ec7e600-0dfb-4a04-8cb3-4ab17cf04c98.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc5NzA3NTQsIm5iZiI6MTcwNzk3MDQ1NCwicGF0aCI6Ii8xNTYwMzMwMTQvMzA0ODQ5NzgwLTRlYzdlNjAwLTBkZmItNGEwNC04Y2IzLTRhYjE3Y2YwNGM5OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIxNVQwNDE0MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wZDMxNTg0ZDRiYWUyZTk1MzYxNjI2YTEwNmQ0N2ExODM2YmZkNGQ4ZjgzNmY2NDYzMTBjMzQ4Y2ZhYWRkZjFlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.OhHGghwyAQbT6TTW-TXTFFMttuMK9kGK2L8t26E3g0M)

Is this in a modal as well? Does your modal have some opening animation that changes the height?

**Comment by yuhao-wong** [comment]
Yes its in a modal, but I am rendering it with [React.createPortal](https://react.dev/reference/react-dom/createPortal).
No animation.

**Comment by hanurii** [comment]
In my case, the issue was resolved when I didn't provide initial values in the crop's useState.

```js
const [crop, setCrop] = useState();
```

**Comment by ak-pdeshaies** [comment]
THANK YOU!!!

Both proposed solution found here worked for me
1) `document.querySelector(".ReactCrop__crop-mask")?.setAttribute("height", "100%")`
2) Remove default crop value `const [crop, setCrop] = useState();`

I ended up using option 2. FYI, if you need to specify a crop do it on imageLoad with setCrop({ ... }) That's how I'm doing it and I fixed the issue I was having. \o/
